src/java.base/windows/native/libjava/FileDescriptor_md.c

```diff
@@ -94,6 +97,54 @@ Java_java_io_FileCleanable_cleanupClose0(JNIEnv *env, jclass fdClass, jint unuse
 }
 
 JNIEXPORT jstring JNICALL
-Java_java_io_FileDescriptor_nativeDescription0(JNIEnv *env, jobject this) {
-    return (*env)->NewStringUTF(env, "(not implemented)");
-}
\ No newline at end of file
+Java_java_io_FileDescriptor_nativeDescription0(JNIEnv* env, jobject this) {
+    HANDLE handle = (HANDLE)(*env)->GetLongField(env, this, IO_handle_fdID);
+    #define BufferSize 1024
+    char lpszFilePath[BufferSize] = {'\0'};
+
+    HMODULE hKernelDll = LoadLibrary(TEXT("kernel32.dll"));
+    if (!hKernelDll) {
+        JNU_ThrowIOExceptionWithLastError(env, "LoadLibrary kernel32.dll failed");
+        return NULL;
+    }
+
+    HMODULE hNtdllDll = GetModuleHandle(TEXT("ntdll.dll"));
+    if (!hNtdllDll) {
+        JNU_ThrowIOExceptionWithLastError(env, "LoadLibrary ntdll.dll failed");
+        CloseHandle(hKernelDll);
+        return NULL;
+    }
+
+    typedef NTSTATUS(WINAPI* NtQueryObjectFunc)(HANDLE, OBJECT_INFORMATION_CLASS, PVOID, ULONG, PULONG);
+    typedef BOOL(WINAPI* GetFinalPathNameByHandleFunc)(HANDLE, LPSTR, DWORD, DWORD);
```

Comment:
Do you need to load this function dynamically? Contrary to `NtQueryObject` it should be public, present in `fileapi.h`... https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumepathnamesforvolumenamew?redirectedfrom=MSDN

Reply:
Good point, thanks